### PR TITLE
Allow calls to kwarg functions before Base is defined.

### PR DESCRIPTION
**Pull request by maleadt**
Used in #18338. ~~Couldn't find a way to test..~~

### CODE REVIEW

**Comment by vtjnash** [comment]
yes, any `(top)` function should be defined in any top module that might use them, including Core. essentials.jl is a good place for it.

for testing, you could use some variant of:
```julia
julia> eval(Core.Inference, quote
         f(;x...) = println(;x...)
       end)
f (generic function with 1 method)

julia> Core.Inference.f(a=1, b=2, c=3)
ERROR: UndefVarError: vector_any not defined
 in #f#213(::Array{Any,1}, ::Function) at REPL[1]:2
 in (::Core.Inference.#kw##f)(::Array{Any,1}, ::Core.Inference.#f) at <missing>:0
 in eval(::Module, ::Any) at ./boot.jl:234
 in eval(::Module, ::Any) at /Volumes/Lion/Users/jameson/Documents/julia/usr/lib/julia/sys.dylib:?
 in eval_user_input(::Any, ::Base.REPL.REPLBackend) at REPL.jl:66
 in macro expansion; at REPL.jl:97 [inlined]
 in (::Base.REPL.##3#4{Base.REPL.REPLBackend})() at event.jl:68
```

**Comment by maleadt** [comment]
Thanks. Turns out this did work on 0.4, so I've added the regression label.
Rebased and added a test.

File: test/keywordargs.jl [hunk 1]
```diff
@@ -227,3 +227,10 @@ let opts = (:a=>3, :b=>4)
     @test g17785(; opts..., a=5, b=6) == (5, 6)
     @test g17785(; b=0, opts..., a=5) == (5, 4)
 end
+
+# pr #18396, kwargs before Base is defined
+eval(Core.Inference, quote
+    f18396(;kwargs...) = g18396(;kwargs...)
+    g18396(;x=1,y=2) = x+y
+end)
+Core.Inference.f18396()
```
Comment (vtjnash) [inline test/keywordargs.jl]: probably best to  also `@test` the return value so that it is also confirmed to do something

Comment (maleadt) [inline test/keywordargs.jl]: OK, updated.

**Comment by tkelman** [comment]
waiting until 0.5.1

**Pull request merged**